uint8[] belowBlock
bool hasObj
bool openGrip
